app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.java

```diff
@@ -330,6 +359,63 @@ public void selected(ChannelInfoItem selectedItem) {
         importExportListHeader.setOnClickListener(v -> importExportOptions.switchState());
     }
 
+    private void showLongTapDialog(ChannelInfoItem selectedItem) {
+        final Context context = getContext();
+        final Activity activity = getActivity();
+        if (context == null || context.getResources() == null || getActivity() == null) return;
+
+        final String[] commands = new String[]{
+                context.getResources().getString(R.string.share),
+                context.getResources().getString(R.string.unsubscribe)
+        };
+
+        final DialogInterface.OnClickListener actions = (dialogInterface, i) -> {
+            switch (i) {
+                case 0:
+                    shareChannel(selectedItem);
+                    break;
+                case 1:
+                    deleteChannel(selectedItem);
+                    break;
+                default:
+                    break;
+            }
+        };
+
+        final View bannerView = View.inflate(activity, R.layout.dialog_title, null);
+        bannerView.setSelected(true);
+
+        TextView titleView = bannerView.findViewById(R.id.itemTitleView);
+        titleView.setText(selectedItem.getName());
+
+        TextView detailsView = bannerView.findViewById(R.id.itemAdditionalDetails);
+        detailsView.setVisibility(View.GONE);
+
+        new AlertDialog.Builder(activity)
+                .setCustomTitle(bannerView)
+                .setItems(commands, actions)
+                .create()
+                .show();
+
+    }
+
+    private void shareChannel (ChannelInfoItem selectedItem) {
+        shareUrl(selectedItem.getName(), selectedItem.getUrl());
+    }
+
+    @SuppressLint("CheckResult")
+    private void deleteChannel (ChannelInfoItem selectedItem) {
+        ExtractorHelper.getChannelInfo(selectedItem.getServiceId(), selectedItem.getUrl(), true)
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.newThread())
+                .subscribe((@NonNull ChannelInfo result) -> {
+                    List<SubscriptionEntity> toDelete = subscriptionService.subscriptionTable()
+                            .getSubscription(result.getServiceId(), result.getUrl())
+                            .blockingFirst();
+                    subscriptionService.subscriptionTable().delete(toDelete);
+                });
```

Comment:
'subscribe' is also the place where you catch any errors occurring in the async call, so you will also need to implement what happens when any of the logic fails in progress.